src/rt/sched/schedulerthread.h

```diff
@@ -208,6 +208,12 @@ namespace verona::rt
 
         work->run();
 
+        if (work->yielded)
```

Comment:
I don't think you need to modify `schedulerthread.h`.  Also, you don't know that the `work` item is valid after you call run.  It is responsible for its own deallocation.